### ISSUE TITLE
Kahlan 4.1 does not stub PHP functions properly

### ISSUE THREAD

**Issue by lamp-of-god** [kahlan/kahlan#296]
Hi. It seems to be bug with kahlan 4.1

Minimal example:
Code of test file:

```
use Demo\GetContents;

describe('->getContents', function() {

    it('must return correct values', function() {
        allow('file_get_contents')->toBeCalled()->andReturn(false, 'content');
        expect(GetContents::getContents('asd'))->toBe(false);
        expect(GetContents::getContents('asd'))->toBe('content');
    });

    it('must return correct values with doubles', function() {
        allow('file_get_contents')->toBeCalled()->andReturn(false, 'content');
        expect(GetContents::getContentsDouble('asd'))->toBe('content');
    });

});
```

Code of `GetContents` file:

```
<?php

namespace Demo;


class GetContents
{
    public static function getContents($url)
    {
        return file_get_contents($url);
    }

    public static function getContentsDouble($url)
    {
        $value1 = file_get_contents($url);
        return file_get_contents($url);
    }
}
```

In Kahlan v4.0.6 both of tests are passed. In Kahlan 4.1+ only first test is passed.

You can see that in first function `file_get_contents` is called one time, and in second - two times.
Hovewer, if you will invoke `GetContents::getContents` multiple times it will return different (correct) values.

So the problem seems to be with multiple calls of stubbed function in one method.

Source to test:

[kahlan-demo.zip](https://github.com/kahlan/kahlan/files/1871379/kahlan-demo.zip)

**Comment by samsonasik** [comment]
I can confirm it is bug at 4.1.0

**Comment by jails** [comment]
Fixed in 4.1.1
